### DISCUSSION TITLE
Python file association

### DISCUSSION THREAD

**Discussion by doobes** [elementary/triage#547] [Q&A]
So,

I want to do the following with python files (.py) from the native file doohicky in EOS:

1. If I double click on the file, I want to run it in python3
2. If I right click on the file, I'd like to pick a suitable application to open the file, for that instance only (file editor(s), Geany, etc)

How, exactly, do I do this?

Thanks in advance!

**Comment by jeremypw** [comment]
I assume python3 is installed.  In that case the python file needs to have `execute` permission and also begin with the line `#!/usr/bin/env python3`.  You should then get a `run` option in the context menu. Note however that if the python program is not a GUI one but is designed to run in a terminal nothing visible will occur.  A terminal is not automatically launched. I think you need a corresponding `.desktop` file telling the system to launch in a terminal but I haven't looked into the details.
